### ISSUE TITLE
Listing Objects does not return Items

### ISSUE THREAD

**Issue by MarkIvanDev** [TopperDEL/uplink.net#34]
**Describe the bug**
Listing objects inside a bucket does not return Items

**To Reproduce**
1. Get a bucket by calling `var bucket = await bucketService.GetBucketAsync("demo-bucket").ConfigureAwait(false);`
2. List items by calling `var items = await objectService.ListObjectsAsync(bucket, new ListObjectsOptions());`
3. items.Items is empty

**Expected behavior**
Listing items should return the items inside the bucket

**Comment by TopperDEL** [comment]
Listing should definitely work! I'm using it in all of my projects. See [here](https://github.com/TopperDEL/uplink.net/wiki/Documentation#listing-objects).

You may check the following:

- does your access allow listing?
- are you in the correct bucket?
- are there really objects in it?
- are they All in sub-prefixes? If so please check the `recursive` flag in the listing options.

Let me know if you still have issues. It definitely should work.

**Comment by TopperDEL** [comment]
See here how listing is done in duplicati: https://github.com/duplicati/duplicati/blob/184dfeae7860cd34019baf3d3ca635bf6b79dfbe/Duplicati/Library/Backend/Storj/StorjBackend.cs#L273

**Comment by MarkIvanDev** [comment]
Thank you for the response. I have already read the documentation about listing a bucket's items, but I still have no success retrieving the items.

I have also checked the following:
- I have access (by using an access grant) that allows listing. In fact, I have all the permissions set to true.
- Yes, I am in the correct bucket. I am able to retrieve all the buckets inside the project through the library.
- Yes, there are objects inside the bucket. I don't know if this makes a difference but some of my files are less than 4KB. The web console says that *Files less than 4kb are stored as metadata with strong encryption* I have enabled the System and Custom metadata retrieval in the listing options when retrieving object, but still no items.
- The objects are in the root of bucket. I have also tried setting the recursive flag but still it does not return the items.

**Comment by TopperDEL** [comment]
The file size should not matter.

Can you try using uplink-cli and See if you can list your files there? Did you try to upload an object and list it with uplink.net? Please also try to download an object even if it does not Show in listing?

Are there any special Charakters involved? Like Chinese or something?

I still guess it has to do with the access. That you are not allowed to list/encrypt the objects.

**Comment by MarkIvanDev** [comment]
Hi, I have tried creating a new bucket and uploading an item using the cli and uplink.net. When an item is uploaded using the cli or uplink.net, it is being listed when listing using cli and uplink.net. But when a bucket is created using the web interface and items are uploaded using the web interface, the objects are not being listed by the cli and uplink.net. Is this a bug or is this by design? Only the objects uploaded by the cli and uplink.net can be returned when listing? Thank you

There are no special characters, the file names are plain alphanumeric.

I have used the same access grant when using the cli and I can list the items that the cli has uploaded
